Jenkinsfile

```
@@ -205,110 +205,80 @@ stage('Configure') {
 					// The only difference with 7.12+ is that wildcard predicates on analyzed fields get their pattern normalized,
 					// and that was deemed a bug: https://github.com/elastic/elasticsearch/pull/53127
-					new EsLocalBuildEnvironment(versionRange: '[7.11,7.12)', mavenProfile: 'elasticsearch-7.11',
-							condition: TestCondition.ON_DEMAND),
-					new EsLocalBuildEnvironment(versionRange: '[7.12,8.0)', mavenProfile: 'elasticsearch-7.12',
-							condition: TestCondition.AFTER_MERGE),
+					new EsLocalBuildEnvironment(version: '7.11.2', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '7.12.1', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '7.13.2', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '7.16.3', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '7.17.0', condition: TestCondition.AFTER_MERGE),
 					// Not testing 8.0 because we know there are problems in 8.0.1 (see https://hibernate.atlassian.net/browse/HSEARCH-4497)
 					// Not testing 8.1 to make the build quicker.
-					new EsLocalBuildEnvironment(versionRange: '[8.2,8.x)', mavenProfile: 'elasticsearch-8.0',
-							condition: TestCondition.BEFORE_MERGE,
-							isDefault: true),
+					new EsLocalBuildEnvironment(version: '8.1.3', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '8.2.3', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '8.3.3', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '8.4.3', condition: TestCondition.BEFORE_MERGE, isDefault: true),
 
 					// --------------------------------------------
 					// OpenSearch
 					// Not testing 1.0 - 1.2 to make the build quicker.
-					new OpenSearchLocalBuildEnvironment(version: '1.3', mavenProfile: 'opensearch-1.0',
-							condition: TestCondition.AFTER_MERGE),
-					new OpenSearchLocalBuildEnvironment(version: '2.0', mavenProfile: 'opensearch-2.0',
-							condition: TestCondition.ON_DEMAND),
-					new OpenSearchLocalBuildEnvironment(version: '2.1', mavenProfile: 'opensearch-2.1',
-							condition: TestCondition.ON_DEMAND),
-					new OpenSearchLocalBuildEnvironment(version: '2.2', mavenProfile: 'opensearch-2.2',
-							condition: TestCondition.ON_DEMAND),
-					new OpenSearchLocalBuildEnvironment(version: '2.3', mavenProfile: 'opensearch-2.3',
-							condition: TestCondition.AFTER_MERGE)
+					new OpenSearchLocalBuildEnvironment(version: '1.3.6', condition: TestCondition.AFTER_MERGE),
+					new OpenSearchLocalBuildEnvironment(version: '2.0.1', condition: TestCondition.ON_DEMAND),
+					new OpenSearchLocalBuildEnvironment(version: '2.1.0', condition: TestCondition.ON_DEMAND),
+					new OpenSearchLocalBuildEnvironment(version: '2.2.1', condition: TestCondition.ON_DEMAND),
+					new OpenSearchLocalBuildEnvironment(version: '2.3.0', condition: TestCondition.AFTER_MERGE)
```

Let's also move comments from the POM:

```suggestion
					new OpenSearchLocalBuildEnvironment(version: '2.3.0', condition: TestCondition.AFTER_MERGE)
					// See https://opensearch.org/lines/2x.html for a list of all 1.x versions
```